lib/src/match.dart

```diff
@@ -149,10 +149,10 @@ int matchBitap(String text, String pattern, int loc, double threshold,
     int start = max(1, loc - bin_mid + 1);
     int finish = min(loc + bin_mid, text.length) + pattern.length;
 
-    final rd = new List<int>(finish + 2);
+    final rd = new List<int?>.filled(finish + 2, 0);
     rd[finish + 1] = (1 << d) - 1;
     for (int j = finish; j >= start; j--) {
-      int charMatch;
+      int? charMatch;
```

Comment:
I would initialize it with 0 and make it non-nullable.
Additionaly change Line 160 to:
        charMatch = s[text[j - 1]] ?? 0;
So if something would leed to a null in the s[] it would be changed to 0